module/VuFind/src/VuFind/Controller/NotificationsController.php

```diff
@@ -0,0 +1,430 @@
+        }
+
+        $page = [];
+
+        $view = $this->createViewModel();
+        $view->languages = $this->config['Notifications']['languages'];
+
+        if ($page_id != 'NEW') {
+            $page = $pagesTable->getPagesDataByPageId($page_id);
+            $pagesForm->setAttribute(
+                'action',
+                $this->url()->fromRoute('admin/notifications-pages', ['action' => 'EditPage','page_id' => $page_id])
+            );
+            $pagesForm->bind(new ArrayObject($page));
+            $view->page_id = $page_id;
+        } else {
+            $pagesForm->setAttribute(
+                'action',
+                $this->url()->fromRoute('admin/notifications-pages', ['action' => 'EditPage','page_id' => 'NEW'])
+            );
+            $view->addNew = true;
+        }
+
+        $view->form = $pagesForm;
+
+        if (!$this->getRequest()->isPost()) {
+            return $view;
+        }
+
+        $pagesForm->setData($this->getRequest()->getPost());
+        if (!$pagesForm->isValid()) {
+            return $view;
+        }
+
+        $data = $pagesForm->getData();
+        if (!isset($data['author_id']) || $data['author_id'] == '') {
+            $data['author_id'] = $user->id;
+        }
+
+        $pagesTable->insertOrUpdatePage($data, $page, $page_id);
+
+        return $this->redirect()->toRoute('admin/notifications-pages', ['action' => 'Pages']);
+    }
+
+    /**
+     * Edit an exiting or a new broadcast
+     *
+     * @return mixed
+     */
+    public function editBroadcastAction()
+    {
+        if (!stristr($this->getRequest()->getRequestUri(), '/Admin')) {
+            return $this->redirect()->toRoute('admin/notifications-broadcasts', ['action' => 'EditBroadcast']);
+        }
+        if (!$this->getAuthManager()->isLoggedIn() || !$this->permission()->isAuthorized('notifications.Admin')) {
+            return $this->redirect()->toRoute('myresearch-home');
+        }
+        if ($this->getRequest()->getPost('cancel')) {
+            return $this->redirect()->toRoute('admin/notifications-broadcasts', ['action' => 'Broadcasts']);
+        }
+
+        $broadcastsTable = $this->getTable('notifications_broadcasts');
+        $formElementManager = $this->serviceLocator->get('FormElementManager');
+        $broadcastsForm = $formElementManager->get(BroadcastsForm::class);
+
+        $broadcast_id = $this->params()->fromPost('broadcast_id', $this->params()->fromQuery('broadcast_id', []));
+        if (!$broadcast_id) {
+            return $this->redirect()->toRoute('admin/notifications-broadcasts', ['action' => 'Broadcasts']);
+        }
+
+        $broadcast = [];
+
+        $view = $this->createViewModel();
+        $view->languages = $this->config['Notifications']['languages'];
+
+        if ($broadcast_id != 'NEW') {
+            $broadcast = $broadcastsTable->getBroadcastsDataByBroadcastId($broadcast_id);
```

Comment:
After loading $broadcast, I see that the `startdate` and `enddate` elements are datetimes i.e. `2024-11-05 00:00:00`, which makes sense since the schema defines those fields as datetimes.  There is some logic within Laminas form which is corrupting that (I think) so that in FormElement->render(), the $renderedType ends up as `"<input type="date" name="startdate" class="form-control" value="2024-11-05&#x20;00&#x3A;00&#x3A;00">"`, which I think is what messes up the display.

I can fix this right after this line by truncating off the time parts:

```
            foreach(['startdate', 'enddate'] as $dateField) {
                $broadcast[$dateField] = explode(' ', $broadcast[$dateField])[0];
            }
```
But there must be a better way to resolve this.  I'm not sure if the schema has to use datetime in the first place if the UI accepts only date values, unless this is trying to be flexible in case we do store times in the future.

Reply:
@maccabeelevine Thanks for the test! I was about to try this change, but checked the generated HTML on my local system first for the corrupted HTML. And here seems to be the difference to your test and those of @sturkel89 . I get this HTML:

```
<input type="date" name="startdate" class="form-control" value="2024-07-11">
```

So I checked my local database and the schema is `date` not `datetime`. I'm sorry that I missed this before and I can not recall why this setting does not match the schema given in `mysql.sql`!

@demiankatz Will a date be enough, or would you think it should be possible to set exact times? We tested this functionality with two communities (hebis and Qcovery) and I don't think that setting the exact time has been missed.

Reply:
I can imagine there might be arguments for either granularity, but if your existing user base has found date granularity acceptable, it seems reasonable to at least start there. If we need date/time, nothing is stopping us from adding it as a future option. (Of course, if it's trivially easy, maybe it's worth adding a config setting now... but I suspect it is not trivially easy, and I don't want to make the workload here even larger!).

Reply:
I started working on additional time settings, but realized that it will probably be better to get the first version on this feature finalized - assuming that the whole community will have ideas for further development in the future and we will be working on this anyhow.

I changed the settings in mysql.sql from `datetime` to `time` and added the three small changes that @maccabeelevine proposed (although setting `0` as default for `priority`).